pkgs/development/libraries/libva/default.nix

```diff
@@ -1,43 +1,57 @@
 { stdenv, lib, fetchFromGitHub, autoreconfHook, pkgconfig
 , libXext, libdrm, libXfixes, wayland, libffi, libX11
 , mesa_noglu
-, minimal ? true, libva
 }:
 
-stdenv.mkDerivation rec {
-  name = "libva-${lib.optionalString (!minimal) "full-"}${version}";
-  version = "2.0.0";
+let
+  v1Version = "1.8.3";
+  v1Sum = "0zqbk02h4jl5v0b141wmi5375k2yplklmm6hv4c75aarlxr7vgms";
 
-  src = fetchFromGitHub {
-    owner  = "01org";
-    repo   = "libva";
-    rev    = version;
-    sha256 = "1x8rlmv5wfqjz3j87byrxb4d9vp5b4lrrin2fx254nwl3aqy15hy";
-  };
+  v2Version = "2.0.0";
+  v2Sum = "1x8rlmv5wfqjz3j87byrxb4d9vp5b4lrrin2fx254nwl3aqy15hy";
+
+  generic = version: sha256: minimal: stdenv.mkDerivation rec {
+    name = "libva-${lib.optionalString (!minimal) "full-"}${version}";
+
+    src = fetchFromGitHub {
+      owner  = "01org";
+      repo   = "libva";
+      rev    = version;
+      inherit sha256;
+    };
 
-  outputs = [ "dev" "out" ];
+    outputs = [ "dev" "out" ];
 
-  nativeBuildInputs = [ autoreconfHook pkgconfig ];
+    nativeBuildInputs = [ autoreconfHook pkgconfig ];
 
-  buildInputs = [ libdrm ]
-    ++ lib.optionals (!minimal) [ libva libX11 libXext libXfixes wayland libffi mesa_noglu ];
-  # TODO: share libs between minimal and !minimal - perhaps just symlink them
+    buildInputs = [ libdrm ]
+      ++ lib.optionals (!minimal) [ libX11 libXext libXfixes wayland libffi mesa_noglu ];
+    # TODO: share libs between minimal and !minimal - perhaps just symlink them
 
-  enableParallelBuilding = true;
+    enableParallelBuilding = true;
 
-  configureFlags = [
-    "--with-drivers-path=${mesa_noglu.driverLink}/lib/dri"
-  ] ++ lib.optionals (!minimal) [ "--enable-glx" ];
+    configureFlags = [
+      "--with-drivers-path=${mesa_noglu.driverLink}/lib/dri"
+    ] ++ lib.optionals (!minimal) [ "--enable-glx" ];
 
-  installFlags = [
-    "dummy_drv_video_ladir=$(out)/lib/dri"
-  ];
+    installFlags = [
+      "dummy_drv_video_ladir=$(out)/lib/dri"
+    ];
 
-  meta = with stdenv.lib; {
-    description = "VAAPI library: Video Acceleration API";
-    homepage = http://www.freedesktop.org/wiki/Software/vaapi;
-    license = licenses.mit;
-    maintainers = with maintainers; [ garbas ];
-    platforms = platforms.unix;
+    passthru = { inherit version; };
```

Comment:
I would personally prefer to inherit it directly (i.e. pass it inside like `name`):
 - it's typically done in nixpkgs nowadays,
 - it avoids yet another mass rebuild.